src/System.Net.Http.WinHttpHandler/tests/UnitTests/FakeInterop.cs

```diff
@@ -611,6 +611,24 @@ private static bool CopyToBufferOrFailIfInsufficientBufferLength(string value, I
             return true;
         }
 
+        public static bool WinHttpDetectAutoProxyConfigUrl(
```

Comment:
Since you're adding this method (so that the unit tests keep compiling), it might be useful to also add some new unit tests to test your new code in this PR.

Reply:
Many existing test cases will hit the new code path. (They are exhaustive and cover almost every scenario)

For example, these test cases will hit `WinHttpDetectAutoProxyConfigUrl` returns `true`:

`SendAsync_NoAutomaticProxySupportAndUseWinInetSettingsWithAutoDetectSetting_ExpectedWinHttpProxySettings`
`SendAsync_NoAutomaticProxySupportAndUseWinInetSettingsWithAutoDetectButPACFileNotDetectedOnNetwork_ExpectedWinHttpProxySettings`
`SendAsync_NoAutomaticProxySupportAndUseWinInetSettingsWithAutoDetectSettingAndManualSettingButPACFileNotFoundOnNetwork_ExpectedWinHttpProxySettings`

These test cases will hit `WinHttpDetectAutoProxyConfigUrl` returns `false`:

`SendAsync_NoAutomaticProxySupportAndUseWinInetSettingsWithMissingRegistrySettings_ExpectedWinHttpProxySettings`
`SendAsync_NoAutomaticProxySupportAndUseWinInetSettingsWithEmptySettings_ExpectedWinHttpProxySettings`
`SendAsync_NoAutomaticProxySupportAndUseWinInetSettingsWithManualSettingsOnly_ExpectedWinHttpProxySettings`

The name `NoAutomaticProxySupport` is somewhat confusing, actually it means: when we pass in access type `WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY` to `WinHttpOpen`, the function will return a invalid session handle. (For example, when running on a platform earlier than Win8.1/Win2K12R2). Then we will need to use WPAD related functions to retrieve proxy data.

I think the above 6 test cases covered all proxy setting scenarios we may want to call `WinHttpDetectAutoProxyConfigUrl`.

Reply:
> I think the above 6 test cases covered all proxy setting scenarios we may want to call 

How are you validating the script cases? Are you doing that manually locally by setting up some working auto-config location?

Reply:
> How are you validating the script cases?

There are three cases related to script: 1. discover the PAC URL, 2. download the PAC script file, 3. execute the script and return the proxy configuration. `WinHttpDetectAutoProxyConfigUrl` is only able to achieve case 1: to auto-detect the URL for the proxy auto-config file (without downloading/executing), and we are reply on `WinHttpGetProxyForUrl` to finish case 2 and 3.

So I assume you are asking for the first case, discover the PAC URL. There are two ways to achieve it: 

1. Automatically detect settings (using WPAD protocol, which I think it's hard to setup testing, but @davidsh can confirm?) We can trust `WinHttpDetectAutoProxyConfigUrl` to return the correct PAC URL. (if it exists)
2. Use automatic configuration script. I have tested this scenario and set up a working proxy auto-config location (on a IIS server), the `WinHttpGetIEProxyConfigForCurrentUser` will return the correct URL in this case, and store its value in `AutoConfigUrl`.